src/main/java/org/apache/pirk/querier/wideskies/encrypt/EncryptQuery.java

```diff
@@ -64,9 +67,33 @@
   // Paillier encryption functionality.
   private final Paillier paillier;
 
+  private static final SecureRandom secureRandom;
+
+  static
+  {
+    try
+    {
+      String alg = SystemConfiguration.getProperty("pallier.secureRandom.algorithm");
+      if (alg == null)
+      {
+        secureRandom = new SecureRandom();
+      }
+      else
+      {
+        String provider = SystemConfiguration.getProperty("pallier.secureRandom.provider");
+        secureRandom = (provider == null) ? SecureRandom.getInstance(alg) : SecureRandom.getInstance(alg, provider);
+      }
+      logger.info("Using secure random from " + secureRandom.getProvider().getName() + ":" + secureRandom.getAlgorithm());
+    } catch (GeneralSecurityException e)
+    {
+      logger.error("Unable to instantiate a SecureRandom object with the requested algorithm.", e);
+      throw new RuntimeException("Unable to instantiate a SecureRandom object with the requested algorithm.", e);
+    }
+  }
+
```

Comment:
I have moved the SecureRandom instantiation out into a separate class. It generates a single instance that can be used anywhere by invoking `RandomProvider.getSecureRandom()`. 

[There's a comment explaining why I have concluded that it's safe to share a single SecureRandom object between callers](https://github.com/apache/incubator-pirk/pull/111/files#diff-a67995da937c3aaa748dca7f6d1d176cR62).